plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -372,7 +373,7 @@ def get_metadata(self, metadata_files):
                 and not self.config.override_config.get(importer_constants.KEY_FEED) \
                 and previous_skip_set - current_skip_set == set() \
                 and (self.download_deferred or not missing_units) \
-                and not sync_due_to_unit_removal:
+                and not sync_due_to_unit_removal and not self.force_full:
```

Comment:
i think the force_full should be moved to the beginning of the condition because if it is set to True, we do not care if the revision number has changed or is the same, we just want to make a force re-sync no matter what.